core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
```

Comment:
Honestly I didn't even know that `ancestors` / `state.Path()` was a thing 😃
I'd go with my solution as it seems more "natural" and might simply things in the future.
Why do we need `State.ancestors` anyways? It's only used in a test.